cmd/loadOutputTranslation.go

```diff
@@ -0,0 +1,236 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cmd
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/Azure/azure-storage-azcopy/ste"
+
+	"github.com/Azure/azure-storage-azcopy/common"
+)
+
+const (
+	clfsErrTag            = "ERR"
+	clfsInitializationTag = "AZCOPY-CONFIG:"
+	clfsInfoTag           = "AZCOPY-INFO:"
+	clfsProgressTag       = "AZCOPY:"
+	clfsExitTag           = "AZCOPY-FINAL:"
+)
+
+type clfsExtensionOutputParser struct {
+	lcm          common.LifecycleMgr
+	finishSignal chan interface{} // TODO consider using a wait group
+}
+
+func newClfsExtensionOutputParser(lcm common.LifecycleMgr) *clfsExtensionOutputParser {
+	return &clfsExtensionOutputParser{lcm: lcm, finishSignal: make(chan interface{}, 1)}
+}
+
+// should be started in another go-routine
+func (c *clfsExtensionOutputParser) startParsing(reader io.Reader) {
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if strings.Contains(line, clfsExitTag) {
+			c.processEndOfJob(line)
+		} else if strings.Contains(line, clfsInitializationTag) {
+			c.processInitialization(line)
+		} else if strings.Contains(line, clfsInfoTag) {
+			c.processInfo(line)
+		} else if strings.Contains(line, clfsProgressTag) {
+			c.processProgress(line)
+		} else if strings.Contains(line, clfsErrTag) {
+			c.processError(line)
+		} else {
+			//ignore all other output
+			//c.lcm.Info("ignored: " + line)
+		}
+	}
+
+	c.finishSignal <- ""
+}
+
+func (c *clfsExtensionOutputParser) finishParsing() {
+	<-c.finishSignal
+}
+
+func (c *clfsExtensionOutputParser) processInitialization(line string) {
+	cleanMessage := strings.TrimPrefix(line, clfsInitializationTag)
+
+	initInfo := clfsInitialization{}
+	err := json.Unmarshal([]byte(cleanMessage), &initInfo)
+	if err != nil {
+		// don't crash if these info are not critical and are only nice-to-have
+		return
+	}
+
+	c.lcm.Info(fmt.Sprintf("CLFSLoad Extension version: %s", initInfo.Version))
+	c.lcm.Info(fmt.Sprintf("CLFSLoad Extension configurations: compression type=%s, preserve hard links=%v",
+		initInfo.Compression, initInfo.PreserveHardLinks))
+
+	if initInfo.New {
+		c.lcm.Info("Starting a new job.")
+	}
+}
+
+func (c *clfsExtensionOutputParser) processInfo(line string) {
+	cleanMessage := strings.TrimPrefix(line, clfsInfoTag)
+
+	info := clfsInfo{}
+	err := json.Unmarshal([]byte(cleanMessage), &info)
+	if err != nil {
+		// don't crash if these info are not critical and are only nice-to-have
+		return
+	}
+
+	c.lcm.Info(fmt.Sprintf("%s phase has started.", info.Phase))
+}
+
+func (c *clfsExtensionOutputParser) processError(line string) {
```

Comment:
Do errors also get logged somewhere?  E.g. by the Python code, to a log file?

Reply:
Yes! it's in the state path.